polar-core/src/polar.rs

```diff
@@ -249,6 +249,16 @@ impl Polar {
             kb.clear_rules();
             return Err(e);
         }
+
+        // TODO(@gkaemmer) is this the right place for this?
+        let mut warnings = vec![];
+        warnings.append(&mut check_no_allow_rule(&*kb)?);
+        let warnings = warnings.into_iter().map(|msg| Message {
+            kind: MessageKind::Warning,
+            msg,
+        });
```

Comment:
I like the idea of `Message::warning` -- I think I'll add that.

On the second comment, I think I'll keep the method signature as-is for now. Right now, our warning methods (in `warnings.rs`) all return `PolarResult<Vec<String>>`, so this would probably mean changing all of them. That's fine, but IMO it'd be even more ideal if these methods would each return a list of specific `Warning` or `Error` objects, which the caller would then gather up and convert into messages. A couple benefits to this approach:
- I imagine it'd be easy to "promote" warnings to errors (like we did with singleton warnings -> errors)
- With a standard for warning types, users could tell Oso to ignore certain warnings
- We could use the same pipeline for validating policies during `load_files` and also in an LSP context, where the output format is different (likely not using the `Message` type).

Anyway, I feel like there's a chunk of refactoring work here to do, but don't want to do it in this change. For now, I think keeping the same method signature as the other `check_*` methods (`PolarResult<Vec<String>>`) is the simplest thing to do.